### PR TITLE
sync-diff-inspector: support checkpoint

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
sync-diff-inspector: support checkpoint
update log pkg to pingcap/log by the way

### What is changed and how it works?
cherry pick #224 
cherry pick #217 
resolve some conflicts.

### Check List 

Tests 

 - Unit test

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
LGTM

**Comment by csuzhangxc** [comment]
do we _sync log before exit_ for binaries? ref https://github.com/pingcap/tidb-lightning/pull/168

File: go.mod1 [hunk 1]
```diff
@@ -3,87 +3,53 @@ module github.com/pingcap/tidb-tools
...
 	github.com/grpc-ecosystem/go-grpc-middleware v1.0.0 // indirect
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0 // indirect
 	github.com/grpc-ecosystem/grpc-gateway v1.5.1 // indirect
-	github.com/jonboulle/clockwork v0.1.0 // indirect
 	github.com/klauspost/cpuid v0.0.0-20180405133222-e7e905edc00e // indirect
 	github.com/konsorten/go-windows-terminal-sequences v1.0.1 // indirect
 	github.com/kr/pretty v0.1.0 // indirect
-	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
 	github.com/montanaflynn/stats v0.0.0-20180911141734-db72e6cae808 // indirect
 	github.com/ngaut/log v0.0.0-20180314031856-b8e36e7ba5ac
```
Comment (csuzhangxc) [inline go.mod1]: where do we use _github.com/ngaut/log_?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

File: go.mod1 (hunk 1 above)
Comment (WangXiangUSTC) [reply go.mod1]: removed

**Comment by WangXiangUSTC** [comment]
@csuzhangxc don't sync log now, and I think it is not emergency, I will add it for every tool later

File: go.mod1 (hunk 1 above)
Comment (csuzhangxc) [reply go.mod1]: :blush: we often forgot to run `go mod tidy` after updated dep.

**Review by csuzhangxc — APPROVED** [review]
LGTM

File: go.mod1 (hunk 1 above)
Comment (WangXiangUSTC) [reply go.mod1]: need add `clean` in makefile

**Pull request merged**